### PR TITLE
Status: Make Host class functions static to simplify their usage

### PR DESCRIPTION
**Pull request by pkuliga**
## Proposed changes:

It makes the `Host` class functions static to simplify their usage.

Instead of:

```php
$is_wpcom_simple = ( new Host() )->is_wpcom_simple();
```

We can just:

```php
$is_wpcom_simple = Host::is_wpcom_simple();
```

This change is backward compatible.

Link to the conversation: p1714460248137239-slack-CBG1CP4EN

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

Just a code review should be fine.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ function ( $options ) {
 	);
 
 	// If called via REST API, we need to register later in the lifecycle
-	if ( ( new Host() )->is_wpcom_platform() && ! jetpack_is_frontend() ) {
+	if ( Host::is_wpcom_platform() && ! jetpack_is_frontend() ) {
```
Comment (pkuliga) [inline projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php:51]: Just an example of usage.

File: projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php [hunk 2]
```diff
@@ -84,7 +84,7 @@ function get_current_url() {
 function get_subscriber_login_url( $redirect ) {
 	$redirect = ! empty( $redirect ) ? $redirect : get_site_url();
 
-	if ( ( new Host() )->is_wpcom_simple() ) {
+	if ( Host::is_wpcom_simple() ) {
```
Comment (pkuliga) [inline projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php:87]: And another one.

**Comment by kraftbj** [comment]
@ice9js @chrisbliss18 Can y'all add this to your team calls' agendas for next week so we can decide if we want to make this change or not?

**Comment by pkuliga** [comment]
Hi @ice9js @chrisbliss18 any decision on this PR?

**Pull request closed without merging**